src/main/java/com/jherrick/GameImpl.java

```diff
@@ -4,76 +4,141 @@
 
 public class GameImpl implements Game {
 
-	private final Board board;
+
+	// ============== Instance variables ================
+	private final BigBoard board;
 	private final Agent xAgent;
 	private final Agent oAgent;
 	private final Side turn;
 
 	private static Scanner reader;
 
-	public GameImpl(Board board, Agent xAgent, Agent oAgent) {
+
+	// ================== Constructors ==================
+
+	public GameImpl(BigBoard board, Agent xAgent, Agent oAgent) {
 		this.board = board;
 		this.xAgent = xAgent;
 		this.oAgent = oAgent;
 		this.turn = Side.X;
 	}
 
-	// main method
+	
+	// ================== Main Method ===================
+
 	public static void initAndPlayGame() {
-		Move m1 = Utils.coordinatesToMove("e5");
-		Move m2 = Utils.coordinatesToMove("e5");
-		System.out.println(m1.equals(m2));
+		
 		reader = new Scanner(System.in);
-
-		// set up the board
-		BigBoard b = new BigBoard();
 		
-		// get input
-		System.out.println("Welcome. Do you want a custom position?");
-		String s = reader.nextLine();
-		if (s.equals("yes")) {
-			b = inputCustomBoard();
-		}
+		BigBoard b = askForCustomBoard(reader);
+		Agent [] agents = assignAgentType(reader, b);
 
-		Agent xAgent = new ConsolePlayerAgent(reader);
-		Agent oAgent = new ConsolePlayerAgent(reader);
-
-		Game game = new GameImpl(b, xAgent, oAgent);
+		Game game = new GameImpl(b, agents[0], agents[1]);
+		
 		// use the inputed board
 		game.play();
 		
 		// close scanner
 		reader.close();
 	}
 
-	// input a particular board to play
-	public static BigBoard inputCustomBoard() {
-		System.out.println("Please enter the HCN string for the custom board.");
-		String s = reader.nextLine();
-		BigBoard b = new BigBoard(s);
-		return b;
-	}
 
 	@Override
 	public void play() {
 		boolean gameOver = false;
 		while(!gameOver){
 			System.out.println(board);
+			// TODO: Bug - Fix CPU making all moves immediately
 			Move nextMove = getNextMove();
+
+			// Exit game if move is -1
+			if (nextMove.move == -1){
+				break;
+			}
+
 			this.board.makeMove(nextMove);
-			gameOver = this.board.getBoardState() != BoardState.IN_PROGRESS;
+			// TODO: Fix bug, no Game Over when BigBoard is won
+			gameOver = this.board.getBoardState() != BoardState.IN_PROGRESS; 
 		}
 		System.out.println("Game Over!");
 	}
 
+
+
+
+	// ================ Helper Functions =================
+
+	// ======== Move Generation Helper Functions =========
+	
 	private Move getNextMove() {
 		if(this.turn.equals(Side.X)){
 			return xAgent.pickMove(board);
 		}
 		else return oAgent.pickMove(board);
 	}
 
-	private void gameFlowToBeUpdated(){
+	// ======= HCN Custom Board Helper Functions ============
+
+	private static BigBoard askForCustomBoard(Scanner reader){
```

Comment:
I think this method should be called initBoard or something. And then it could branch into initCustomBoard and initDefaultBoard.